src/affine3d.rs

```diff
@@ -0,0 +1,639 @@
+    ///
+    /// The resulting matrix can be used to transform 3D points and vectors. See
+    /// [`Self::transform_point3()`] and [`Self::transform_vector3()`].
+    #[inline(always)]
+    pub fn from_translation(translation: Vec3) -> Self {
+        Self::from_rows(
+            Vec4::new(1.0, 0.0, 0.0, translation.x),
+            Vec4::new(0.0, 1.0, 0.0, translation.y),
+            Vec4::new(0.0, 0.0, 1.0, translation.z),
+        )
+    }
+
+    /// Creates an affine transformation matrix from the given 3D `scale`, `rotation` and
+    /// `translation`.
+    ///
+    /// Equivalent to `Affine3D::from_translation(translation) * Affine3D::from_quat(rotation) * Affine3D::from_scale(scale)`
+    ///
+    /// The resulting matrix can be used to transform 3D points and vectors. See
+    /// [`Self::transform_point3()`] and [`Self::transform_vector3()`].
+    #[inline(always)]
+    pub fn from_scale_rotation_translation(scale: Vec3, rotation: Quat, translation: Vec3) -> Self {
+        let scale4 = scale.extend(0.0);
+        let mut m = Self::from_quat(rotation);
+        m.x_row *= scale4;
+        m.y_row *= scale4;
+        m.z_row *= scale4;
+        m.x_row.w = translation.x;
+        m.y_row.w = translation.y;
+        m.z_row.w = translation.z;
+        m
+    }
+
+    /// Creates an affine transformation matrix from the given 3D `translation`.
+    ///
+    /// Equivalent to `Affine3D::from_translation(translation) * Affine3D::from_quat(rotation)`
+    ///
+    /// The resulting matrix can be used to transform 3D points and vectors. See
+    /// [`Self::transform_point3()`] and [`Self::transform_vector3()`].
+    #[inline(always)]
+    pub fn from_rotation_translation(rotation: Quat, translation: Vec3) -> Self {
+        let mut m = Self::from_quat(rotation);
+        m.x_row.w = translation.x;
+        m.y_row.w = translation.y;
+        m.z_row.w = translation.z;
+        m
+    }
+
+    /// The given `Mat4` must be an affine transform,
+    /// i.e. contain no persepctive transform.
+    #[inline]
+    pub fn from_mat4(m: Mat4) -> Self {
+        Self::from_cols(
+            m.x_axis.truncate(),
+            m.y_axis.truncate(),
+            m.z_axis.truncate(),
+            m.w_axis.truncate(),
+        )
+    }
+
+    /// The first column.
+    #[inline(always)]
+    fn x_col(&self) -> Vec3 {
+        Vec3::new(self.x_row.x, self.y_row.x, self.z_row.x)
+    }
+
+    /// The second column.
+    #[inline(always)]
+    fn y_col(&self) -> Vec3 {
+        Vec3::new(self.x_row.y, self.y_row.y, self.z_row.y)
+    }
+
+    /// The thirs column.
+    #[inline(always)]
+    fn z_col(&self) -> Vec3 {
+        Vec3::new(self.x_row.z, self.y_row.z, self.z_row.z)
+    }
+
+    /// The translation expressed by this transform.
+    /// The translation is applied last, so is separatable from scale, shear and rotation.
+    #[inline(always)]
+    pub fn translation(&self) -> Vec3 {
```

Comment:
It might also be good to provide accessors to basis vectors and possibly the rotation/scale 3x3 matrix, although this could happen separate to this PR.

Reply:
I wanted to keep the interface a bit more implementation agnostic for now, but setting/getting the full `Mat3` would be useful.

Reply:
Added setter/getter for `Mat3`